suite-native/storage/src/storage.ts

```diff
@@ -1,23 +1,44 @@
 import { MMKV } from 'react-native-mmkv';
 
+import * as Random from 'expo-random';
+import * as SecureStore from 'expo-secure-store';
 import { Storage } from 'redux-persist';
 
-const storage = new MMKV({
-    id: 'trezorSuite-app-storage',
-    encryptionKey: 'FIXME',
-});
-
-export const mmkvStorage: Storage = {
-    setItem: (key, value) => {
-        storage.set(key, value);
-        return Promise.resolve(true);
-    },
-    getItem: key => {
-        const value = storage.getString(key);
-        return Promise.resolve(value);
-    },
-    removeItem: key => {
-        storage.delete(key);
-        return Promise.resolve();
-    },
+const ENCRYPTION_KEY = 'STORAGE_ENCRYPTION_KEY';
+
+export const purgeStorageEncryptionKey = () => SecureStore.deleteItemAsync(ENCRYPTION_KEY);
+
+const retrieveStorageEncryptionKey = async () => {
+    let secureKey = await SecureStore.getItemAsync(ENCRYPTION_KEY);
+
+    if (secureKey == null) {
+        secureKey = Buffer.from(Random.getRandomBytes(32)).toString('hex');
+        await SecureStore.setItemAsync(ENCRYPTION_KEY, secureKey);
+    }
+
+    return secureKey;
+};
+
+export const initMmkvStorage = async () => {
+    const encryptionKey = await retrieveStorageEncryptionKey();
+
+    const storage = new MMKV({
+        id: 'trezorSuite-app-storage',
+        encryptionKey,
+    });
+
+    return {
+        setItem: (key, value) => {
+            storage.set(key, value);
+            return Promise.resolve(true);
+        },
+        getItem: key => {
+            const value = storage.getString(key);
+            return Promise.resolve(value);
+        },
+        removeItem: key => {
+            storage.delete(key);
+            return Promise.resolve();
+        },
+    } as Storage;
```

Comment:
Do you need type casting? Is adding return type not enough?

Reply:
bd8e22ff3